src/analyzers/piebald.rs

```diff
@@ -0,0 +1,350 @@
+
+    let chats = stmt
+        .query_map([], |row| {
+            Ok(PiebaldChat {
+                id: row.get(0)?,
+                title: row.get(1)?,
+                model: row.get(2)?,
+                current_directory: row.get(3)?,
+            })
+        })?
+        .filter_map(|r| r.ok())
+        .collect();
+
+    Ok(chats)
+}
+
+/// Query all messages from the database.
+fn query_messages(conn: &Connection) -> Result<Vec<PiebaldMessage>> {
+    let mut stmt = conn.prepare(
+        "SELECT id, parent_chat_id, role, input_tokens, output_tokens,
+                reasoning_tokens, cache_read_tokens, cache_write_tokens, created_at
+         FROM messages
+         ORDER BY created_at",
+    )?;
+
+    let messages = stmt
+        .query_map([], |row| {
+            Ok(PiebaldMessage {
+                id: row.get(0)?,
+                parent_chat_id: row.get(1)?,
+                role: row.get(2)?,
+                input_tokens: row.get(3)?,
+                output_tokens: row.get(4)?,
+                reasoning_tokens: row.get(5)?,
+                cache_read_tokens: row.get(6)?,
+                cache_write_tokens: row.get(7)?,
+                created_at: row.get(8)?,
+            })
+        })?
+        .filter_map(|r| r.ok())
+        .collect();
+
+    Ok(messages)
+}
+
+/// Parse a timestamp string from Piebald's database.
+///
+/// Piebald stores timestamps in RFC3339 format with timezone (e.g., "2025-12-10T15:55:48.819321712+00:00").
+/// Returns None if the timestamp cannot be parsed.
+fn parse_timestamp(ts: &str) -> Option<DateTime<Utc>> {
+    // Try RFC3339 first (with timezone) - this is Piebald's format
+    if let Ok(dt) = DateTime::parse_from_rfc3339(ts) {
+        return Some(dt.with_timezone(&Utc));
+    }
+
+    // Try SQLite's default datetime format as fallback
+    if let Ok(naive) = NaiveDateTime::parse_from_str(ts, "%Y-%m-%d %H:%M:%S") {
+        return Some(naive.and_utc());
+    }
+
+    // Try with fractional seconds
+    if let Ok(naive) = NaiveDateTime::parse_from_str(ts, "%Y-%m-%d %H:%M:%S%.f") {
+        return Some(naive.and_utc());
+    }
+
+    None
+}
+
+/// Convert Piebald messages to splitrail's ConversationMessage format.
+fn convert_messages(
+    chats: &[PiebaldChat],
+    messages: Vec<PiebaldMessage>,
+) -> Vec<ConversationMessage> {
+    // Build chat lookup map for O(1) access
+    let chat_map: HashMap<i64, &PiebaldChat> = chats.iter().map(|c| (c.id, c)).collect();
+
+    messages
+        .into_iter()
+        .filter_map(|msg| {
+            let chat = chat_map.get(&msg.parent_chat_id)?;
+
+            // Parse timestamp - skip messages with invalid timestamps
+            let date = parse_timestamp(&msg.created_at)?;
+
+            // Use project path from chat's current_directory, falling back to "piebald" if not set
+            let project_hash = hash_text(chat.current_directory.as_deref().unwrap_or("piebald"));
```

Comment:
I don't think the name `piebald` makes sense for a project hash, and Piebald shows Ungrouped for chats without a working directory set, so this seems better to me.
```suggestion
            // Use project path from chat's current_directory, falling back to "ungrouped" if not set
            let project_hash = hash_text(chat.current_directory.as_deref().unwrap_or("ungrouped"));
```